docs/docs/typescript.md

```diff
@@ -0,0 +1,33 @@
+---
+title: TypeScript and Gatsby
+---
+
+## Introductory paragraph
+
+[TypeScript](https://www.typescriptlang.org/) is a JavaScript superset which extends the language to include type definitions allowing codebases to be statically checked for soundness. Gatsby provides an integrated experience out of the box, similar to an IDE.
+
+## Example
+
```

Comment:
Code samples in reference guides are generally accompanied by brief explanations of what's happening in them. In this case, you could say something along the lines of "You can use TypeScript to specify the intended shape of an object. In the following example, the `props` argument that gets passed into the `IndexRoute()` function will be type-checked against `PageProps`."